### PR TITLE
Fix verification of subkeys of default gpg key

### LINKED ISSUE #10309: Instance GPG key not recognized if subkey
- Gitea version (or commit ref): 1.11.0
- Git version: 2.17.2
- Operating system: macOS 10.13.6 (17G11023)
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [x] No
  - [ ] Not relevant
- Log gist:

## Description
Regarding the new signing functionality introduced in #7631

Gitea only checks for the primary key when checking a commit with gpgSettings:
https://github.com/go-gitea/gitea/blob/master/models/gpg_key.go#L738-L742
If one uses a subkey to sign, this doesn't get recognized in the GUI.

## Screenshots
Bildschirmfoto 2020-02-17 um 08 40 17

### PR DESCRIPTION
**Pull request by zeripath**
Fix #10309

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

File: models/gpg_key.go [hunk 1]
```diff
@@ -717,6 +717,8 @@ func verifyWithGPGSettings(gpgSettings *git.GPGSettings, sig *packet.Signature,
 		return commitVerification
 	}
 
+	log.Info("PublicKeyContent: %s", gpgSettings.PublicKeyContent)
```
Comment (CirnoT) [inline models/gpg_key.go]: Too verbose in my opinion

Comment (zeripath) [reply models/gpg_key.go]: Sorry that sneaked in from testing.

```suggestion
```

File: models/gpg_key.go [hunk 2]
```diff
@@ -717,6 +717,7 @@ func verifyWithGPGSettings(gpgSettings *git.GPGSettings, sig *packet.Signature,
 		return commitVerification
 	}
 
+
```
Comment (zeripath) [inline models/gpg_key.go]:
```suggestion
```

**Comment by zeripath** [comment]
I'm suspicious that this may actually fix #10591 too

**Comment by 6543** [comment]
@zeripath can you add a test for this case :)

**Review by 6543 — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]

**Comment by techknowlogick** [comment]
ping LG-TM

**Comment by techknowlogick** [comment]
@zeripath please send backport :)

**Pull request merged**